File: src/chain/examples/idvalidation/DniFormatValidator.java
package chain.examples.idvalidation;

public class DniFormatValidator extends IdValidator {
    @Override protected boolean check(IdDocument doc) {
        return doc.getDni() != null && doc.getDni().matches("\\d{8}");
    }
    @Override protected String name() { return "DniFormatValidator"; }
}

File: src/chain/examples/idvalidation/DniFormatValidator.js
'use strict';

const IdValidator = require('./IdValidator');

class DniFormatValidator extends IdValidator {
  check(doc) { return /^\d{8}$/.test(doc.dni || ''); }
  name()     { return 'DniFormatValidator'; }
}

module.exports = DniFormatValidator;
